akka-http-tests/src/test/scala/akka/http/scaladsl/server/directives/FutureDirectivesSpec.scala

```diff
@@ -123,7 +123,7 @@ class FutureDirectivesSpec extends RoutingSpec with Inside with TestKitBase {
       occurrences = 1,
       message = BasicRouteSpecs.defaultExnHandler500Error("XXX")
     ).intercept {
-        Get() ~> onSuccess(Future.failed(TestException)) { echoComplete } ~> check {
+        Get() ~> onSuccess(Future.failed[Int](TestException)) { echoComplete } ~> check {
```

Comment:
I recall here being some issues with implicit value not being found for onSuccess if a type parameter was not specified (unsure why). I was hoping to use Unit everywhere (as that seemed natural with the exception), but this specific one demanded actual value (again don't remember what the error message was, I'll try to reproduce it and comment)

Reply:
`Unit` doesn't work because the `Tupler` machinery then expects a nullary function instead of `echoComplete`. So, that's fine and as expected.

Having to the type annotation here is also fine, in more realistic use cases the type will be inferred anyway.